core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -56,72 +56,73 @@ public Schema schema() {
     }
   }
 
-  abstract static class BaseFilesTableScan extends BaseMetadataTableScan {
+  private static CloseableIterable<FileScanTask> planFiles(Table table, CloseableIterable<ManifestFile> manifests,
+                                                           Schema tableSchema, Schema projectedSchema,
+                                                           TableScanContext context) {
+    Expression rowFilter = context.rowFilter();
+    boolean caseSensitive = context.caseSensitive();
+    boolean ignoreResiduals = context.ignoreResiduals();
+
+    LoadingCache<Integer, ManifestEvaluator> evalCache = Caffeine.newBuilder().build(specId -> {
```

Comment:
Not related to this change, but my next pr plans to use this logic as well to fix filtering for PartitionsTable.  Maybe at that time will move the evalCache part of this further up to BaseMetadataTables.

Reply:
Do you plan to reuse as-is? If so, maybe, I can move it around in this PR.

Reply:
Yea no problem, I can look at it later, it's part of the code but not the whole thing as PartitionsTable creates another kind of FileScanTask.

Reply:
Let's take a look at it in the PR that will use it then.